LBH.AdultSocialCare.Api/V1/Gateways/Payments/Concrete/PayRunGateway.cs

```diff
@@ -112,11 +112,11 @@ public async Task<int> GetDraftPayRunCount(PayrunType payRunType)
                 .CountAsync();
         }
 
-        public async Task<bool> CheckExistsUnApprovedPayRunAsync(PayrunType payRunType)
+        public async Task<bool> CheckExistsUnApprovedPayRunAsync(PayrunType[] payRunTypes)
         {
             var completedPayRunStatuses = new[] { PayrunStatus.Archived, PayrunStatus.Paid, PayrunStatus.PaidWithHold };
             return await _dbContext.Payruns
-                .Where(pr => pr.Type == payRunType && !completedPayRunStatuses.Contains(pr.Status)).AnyAsync();
+                .Where(pr => payRunTypes.Contains(pr.Type) && !completedPayRunStatuses.Contains(pr.Status)).AnyAsync();
```

Comment:
No need to check payrunType, if there is only one unapproved payrun, it should return false.

Reply:
Updated